testing/mod.ts

```diff
@@ -29,11 +29,95 @@ export function assertEqual(actual: unknown, expected: unknown, msg?: string) {
   }
 }
 
-export function assert(expr: boolean, msg = "") {
+interface Constructor {
+  new (...args: any[]): any;
+}
+
+interface Assert {
+  /** Make an assertion, if not true, then throw. */
+  (expr: boolean, msg?: string): void;
+
+  /** Make an assertion that `actual` and `expected` are equal, deeply. If not
+   * deeply equal, then throw.
+   */
+  equal(actual: unknown, expected: unknown, msg?: string): void;
+
+  /** Make an assertion that `actual` and `expected` are strictly equal.  If
+   * not then throw.
+   */
+  strictEqual(actual: unknown, expected: unknown, msg?: string): void;
+
+  /** Executes a function, expecting it to throw.  If it does not, then it
+   * throws.  An error class and a string that should be included in the
+   * error message can also be asserted.
+   */
+  throws(fn: () => void, errorClass?: Constructor, msgIncludes?: string): void;
+}
+
+export const assert = function assert(expr: boolean, msg = "") {
```

Comment:
why not `export function`

Reply:
Because, as you realised below, of the type coercion to allow us to decorate the function safely.